packages/perseus/src/components/highlighting/ui/highlighting-ui.tsx

```diff
@@ -61,7 +61,9 @@ class HighlightingUI extends React.PureComponent<HighlightingUIProps> {
         }
     }
 
-    _selectionIsValid(trackedSelection?: TrackedSelection | null): boolean {
+    _selectionIsValid(
+        trackedSelection?: TrackedSelection | null,
+    ): trackedSelection is TrackedSelection {
```

Comment:
This is a type predicate.  This tells TypeScript to set the type of `trackedSelection` to `TrackedSelection` if the function returns true.  See https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates for more details.